### PR TITLE
Updated documentation for zabbix-screen module

### PR DESCRIPTION
**Pull request by okgolove**
Updated module settings.
Do not check screens options because of module screens.

##### SUMMARY

Updated Zabbix screen"module documentation, module options and logic.
The update was discussed here https://github.com/ansible/ansible/pull/56237

Related to #55345

##### ISSUE TYPE

- Docs Pull Request

##### COMPONENT NAME

zabbix_screen

### CODE REVIEW

File: lib/ansible/modules/monitoring/zabbix/zabbix_screen.py [hunk 1]
```diff
@@ -316,7 +353,16 @@ def main():
             http_login_password=dict(type='str', required=False, default=None, no_log=True),
             validate_certs=dict(type='bool', required=False, default=True),
             timeout=dict(type='int', default=10),
-            screens=dict(type='list', required=True)
+            screens=dict(type='list', elements='dict', required=True, options=dict(
+                screen_name=dict(type='str', required=True),
+                host_group=dict(type='str'),
+                state=dict(type='str', default='present', choices=['absent', 'present']),
+                graph_names=dict(type='list', elements='str'),
+                graph_width=dict(type='int', default=200),
```
Comment (felixfontein) [inline lib/ansible/modules/monitoring/zabbix/zabbix_screen.py]: The default values of `graph_width` and `graph_height` in the code used to be `None`, not 200 resp. 100. Is this change intended? I don't know how the settings work internally (didn't bother to check), but such a change could modify the behavior of the module (if `None` was "keep whatever the user currently had configured").

File: lib/ansible/modules/monitoring/zabbix/zabbix_screen.py [hunk 2]
```diff
@@ -31,15 +31,52 @@
...
+                description:
+                    - Host group will be used for searching hosts.
+                    - Required if I(state=present).
+                type: str
+                required: yes
+            state:
+                description:
+                    - I(present) - Create a screen if it doesn't exist. If the screen already exists, the screen will be updated as needed.
+                    - I(absent) - If a screen exists, the screen will be deleted.
+                type: str
```
Comment (felixfontein) [inline lib/ansible/modules/monitoring/zabbix/zabbix_screen.py:53]: You need `choices` here, too.

File: lib/ansible/modules/monitoring/zabbix/zabbix_screen.py (hunk 1 above)
Comment (okgolove) [reply lib/ansible/modules/monitoring/zabbix/zabbix_screen.py]: I guess it may cause an unexpected behaviour for users who already using the module.
So, we should use `None`.

**Review by D3DeFi — APPROVED** [review]
@okgolove thank you very much for making this! :) I think it is good to go
shipit

**Comment by D3DeFi** [comment]
Can you please squash commits before we merge this PR @okgolove?

**Comment by okgolove** [comment]
@D3DeFi squashed.

**Comment by D3DeFi** [comment]
+1

**Review by felixfontein — APPROVED** [review]
shipit

File: lib/ansible/modules/monitoring/zabbix/zabbix_screen.py [hunk 3]
```diff
@@ -31,15 +31,55 @@
...
             - List of screens to be created/updated/deleted (see example).
-            - If a screen has already been added, the screen name won't be updated.
-            - When creating or updating a screen, C(screen_name) and C(host_group) are required.
-            - When deleting a screen, the C(screen_name) is required.
-            - Option C(graphs_in_row) will limit columns of a screen and make multiple rows (default 3).
-            - >
-              The available states are: C(present) (default) and C(absent). If the screen already exists, and the state is not C(absent), the screen
-              will be updated as needed.
-            - To sort hosts alphabetically, set the C(sort) option to C(true)
-        required: true
```
Comment (resmo) [inline lib/ansible/modules/monitoring/zabbix/zabbix_screen.py]: I think `required: true` should be kept, shouldn't it?

Comment (felixfontein) [reply lib/ansible/modules/monitoring/zabbix/zabbix_screen.py]: Yes, definitely! I guess it got removed accidentally...

**Review by felixfontein — CHANGES_REQUESTED** [review]
`required: true` is missing, see @resmo's comment

File: lib/ansible/modules/monitoring/zabbix/zabbix_screen.py (hunk 3 above)
Comment (okgolove) [reply lib/ansible/modules/monitoring/zabbix/zabbix_screen.py]: It did. I've fixed it.

**Review by resmo — APPROVED** [review]
LGTM

**Review by felixfontein — APPROVED** [review]
shipit

**Comment by felixfontein** [comment]
@okgolove thanks for fixing this! I think this really improves the module :)
@D3DeFi @resmo thanks for reviewing this!
